python/dolfinx/fem/forms.py

```diff
@@ -157,10 +157,15 @@ def _form(form):
         constants = [c._cpp_object for c in form.constants()]
 
         # Subdomain markers (possibly None for some dimensions)
-        subdomains = {_cpp.fem.IntegralType.cell: subdomains.get("cell"),
-                      _cpp.fem.IntegralType.exterior_facet: subdomains.get("exterior_facet"),
-                      _cpp.fem.IntegralType.interior_facet: subdomains.get("interior_facet"),
-                      _cpp.fem.IntegralType.vertex: subdomains.get("vertex")}
+        def unwrap_mt(t):
+            try:
+                return t._cpp_object
+            except AttributeError:
+                return t
+        subdomains = {_cpp.fem.IntegralType.cell: unwrap_mt(subdomains.get("cell")),
+                      _cpp.fem.IntegralType.exterior_facet: unwrap_mt(subdomains.get("exterior_facet")),
+                      _cpp.fem.IntegralType.interior_facet: unwrap_mt(subdomains.get("interior_facet")),
+                      _cpp.fem.IntegralType.vertex: unwrap_mt(subdomains.get("vertex"))}
```

Comment:
Could we for sake of clarity, call this variable something else than `subdomains` (as we use subdomains for the inner dictionary here as well)?